orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/servergroup/strategies/MonitoredDeployStrategy.groovy

```diff
@@ -421,12 +410,74 @@ class MonitoredDeployStrategy implements Strategy {
     return stages
   }
 
+  List<Stage> composeRollbackStages(Stage parent) {
+    CreateServerGroupStage.StageData stageData = parent.mapTo(CreateServerGroupStage.StageData)
+    MonitoredDeployStageData monitoredDeployStageData = parent.mapTo(MonitoredDeployStageData)
+    String deployedServerGroupName = stageData.getServerGroup()
+
+    // Does the user want an automatic rollback?
+    if ((monitoredDeployStageData.failureActions.rollback != FailureActions.RollbackType.Automatic) &&
+      (monitoredDeployStageData.failureActions.rollback != FailureActions.RollbackType.Manual)) {
+      log.warn("Not performing automatic rollback on failed deploy of ${deployedServerGroupName ?: '<NO SERVER GROUP CREATED>'} because no rollback was requested by user in pipeline config")
+      return Collections.emptyList()
+    }
+
+    if (!deployedServerGroupName) {
+      // did not get far enough to create a new server group
+      log.warn("Not performing automatic rollback because the server group was not created")
+      return Collections.emptyList()
+    }
+
+    List<Stage> stages = new ArrayList<>()
+
+    stages << newStage(
+      parent.execution,
+      RollbackClusterStage.PIPELINE_CONFIG_TYPE,
+      "Rollback ${stageData.cluster}",
+      [
+        credentials              : stageData.credentials,
+        cloudProvider            : stageData.cloudProvider,
+        regions                  : [stageData.region],
+        serverGroup              : stageData.serverGroup,
+        stageTimeoutMs           : MINUTES.toMillis(30), // timebox a rollback to 30 minutes
```

Comment:
It's a bit arbitraty, but i reused/copy pasted from https://github.com/spinnaker/orca/blob/master/orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/servergroup/CreateServerGroupStage.groovy#L133

The idea here is that we want to make sure that we don't stay in this "bad" state for long (bad being the deploy failed and we are rolling back, normally the rollback should take a few minutes at most). If the rollback times out the user will (hopefully) be notified because the pipeline will fail. (assuming they have notifications turned on)